torch/csrc/api/include/torch/nn/init.h

```diff
@@ -21,16 +24,42 @@ enum class Nonlinearity {
   LeakyReLU
 };
 
+// This enum class is deprecated and will be removed in 1.5.
```

Comment:
Hmm, I wonder if we can *statically* warn in these cases... does C10_DEPRECATED work here?

Reply:
It seems that adding `C10_DEPRECATED` prevents the enum class from being used in some CI configs such as `pytorch_macos_10_13_py3_test`, hence breaking BC-support. For now I'll keep the comment instead.